### PR TITLE
Gutenberg 7.3.0: Trigger E2E tests

### PR DESCRIPTION
**Pull request by chriskmnds**
E2E integration tests against Gutenberg Edge.

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss [hunk 1]
```diff
@@ -65,10 +65,10 @@
 		border-color: rgba( #425863, 0.4 ); // Gutenberg $dark-opacity-light-800
 	}
 
-	.editor-block-list__block:first-child & {
+	.block-editor-block-list__block:first-child & {
```
Comment (mmtr) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: This will require a new FSE release @Automattic/cylon

Comment (noahtallen) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: If the CSS class is changing, it might be a good idea to keep the old one to handle the case where a user is still on an older Gutenberg version (unless this Gutenberg update has already been pushed to everyone on simple and atomic)

Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: I believe these refactors happened a while back in Gutenberg. This seems like the relevant PR: https://github.com/WordPress/gutenberg/pull/14420

Comment (noahtallen) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: Ah yeah, removing legacy support would certainly do it. Thanks for adding the fix! By the way, how did you find that these broke?

Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: > how did you find that these broke 

Not sure I recall exactly, but probably something I spotted while looking into some of the e2e test failures. This in particular may have led me to it: https://github.com/Automattic/wp-calypso/pull/38857/files#diff-926c3450baa62272252cb16253634da9L27-R27

File: apps/o2-blocks/src/todo/editor.scss [hunk 2]
```diff
@@ -135,7 +135,7 @@ $color-accent-green: #31843f;
 }
 
 // Same for the "add item" input.
-.editor-block-list__block .wp-block-a8c-todo .add-new-item {
+.block-editor-block-list__block .wp-block-a8c-todo .add-new-item {
```
Comment (chriskmnds) [inline apps/o2-blocks/src/todo/editor.scss]: @mmtr anyone to nudge about this? 😀

Comment (mmtr) [reply apps/o2-blocks/src/todo/editor.scss]: I don't know who owns o2 blocks, but they are used in P2s so we can assume that latest Gutenberg will be only available. This will also require to generate a new build and upload it to WP.com but I think @ockham is familiar with the process.

File: test/e2e/lib/gutenberg/gutenberg-editor-component.js [hunk 3]
```diff
@@ -225,7 +235,7 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 				.toLowerCase() }`
 		);
 		const insertedBlockSelector = By.css(
-			`.block-editor-block-list__block.is-selected[aria-label*='Block: ${ name }']`
+			`.block-editor-block-list__block.${ selectedBlockConfirmClass }[aria-label*='${ name } Block']`
```
Comment (ockham) [inline test/e2e/lib/gutenberg/gutenberg-editor-component.js]: What's going on here is that we're inserting a block that contains child blocks. When normally the entire block would then be selected, it's only the first child block in this case:

pricing table

Ideally, we'll find a way to make our selector accomodate for both cases :thinking: 

If we can't, I'd still suggest maybe handling the 'child block selected' case not through a generic-sounding `selectedBlockConfirmClass`, but maybe through a bool that determines whether we're dealing with a 'normal' block, or with one that has inner blocks, and set the relevant selector class to either `is-selected` or `has-child-selected`, depending on the value of that bool (so the code represents what's going on maybe a bit more intuitively).

Comment (chriskmnds) [reply test/e2e/lib/gutenberg/gutenberg-editor-component.js]: Yep, sounds like a good idea. I'll do a cleanup in the end to nail things down. :-)

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss (hunk 1 above)
Comment (mmtr) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: We might still need to keep the old `.editor-` selectors in order to support old versions of Gutenberg. In any case, since these changes are unrelated to the E2E fixes, I'd move them to a separate PR.

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss [hunk 4]
```diff
@@ -406,7 +406,7 @@ body.admin-bar:not( .is-fullscreen-mode ) .page-template-modal-screen-overlay {
 			> .editor-inner-blocks
 			> .editor-block-list__layout
 			> [data-type='core/column']
-			> .editor-block-list__block-edit
```
Comment (mmtr) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: Same as above (let's keep the old `.editor-` selectors and move these changes to a separate PR).

File: apps/o2-blocks/src/todo/editor.scss (hunk 2 above)
Comment (mmtr) [reply apps/o2-blocks/src/todo/editor.scss]: Same as above (let's keep the old `.editor-` selectors and move these changes to a separate PR).

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss (hunk 1 above)
Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: Ok, I can create a new PR with these changes.

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss (hunk 4 above)
Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: Hmmm I wonder if this is actually something we should be concerned at all here. I can see several other occurrences of the non-legacy/newer formats in this and other files in the plugin. I feel a more concentrated effort ought to be carried out for this type of thing? There's at least 20+ occurrences of `block-editor-block` suffixed classes in the plugin that could potentially be candidates for legacy-support. 🤔

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss (hunk 1 above)
Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/style.scss]: See my comment here please before I proceed: https://github.com/Automattic/wp-calypso/pull/38857/files#r369510379 I may be missing something.

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss (hunk 4 above)
Comment (mmtr) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: I'm not familiar with what's the backwards compatibility policy we have on the FSE plugin, but I think we tried to have it compatible with at least the 2 latest releases of WP Core (so FSE can work even if the Gutenberg plugin is not active). So I'd assume that any existing selector is currently working with WP 5.2 and we shouldn't worry about them. @Automattic/cylon or @Automattic/ajax might know more.

Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: Ok. My argument is just that if `block-editor-block__x` won't work here due to missing support for older browsers, then it won't work anywhere. I am not sure if it makes a lot of sense to add support here and leave it hanging loose elsewhere. I feel a more holistic effort would be appropriate.

Comment (obenland) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: I'd defer to @noahtallen for a final call on this, but personally I don't think we'd need to be too worried about WP back compat. The plugin is meant to work with simple and atomic, third-party users are installing it on their own risk. See https://github.com/Automattic/wp-calypso/blob/master/apps/full-site-editing/full-site-editing-plugin/readme.txt#L32

Comment (noahtallen) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: +1 for what @obenland said. We require the latest version of the Gutenberg dev plugin in a lot of cases, so we are fine updating everything to the most recent APIs as long as Atomic / Simple sites also have that. 

I think we removed the previous 2 WP version requirement a few months back. I agree with what @chriskmnds is saying about some of the syntax. We've been trying to keep up with newer CSS selectors as we've noticed things breaking, so we've also removed older classnames as well when we've made those updates.

The only case I can think of is covering the gap between 
- FSE plugin update is released with new classnames
- Gutenberg plugin update is released to Simple sites
- Gutenberg plugin update is released to Atomic sites.

Things could break in the period between when one of those is released and all of them is released.

Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: Thanks for the input folks. I believe we are aligned then. 🙂

p.s. it should be safe to apply these refactors to FSE at any point. It's just legacy selectors replaced with the new ones. I will prepare a PR.

Comment (chriskmnds) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: https://github.com/Automattic/wp-calypso/pull/39035

**Review by ockham — APPROVED** [review]
Thanks for this :+1:
